### PR TITLE
Remove step mapconfig overviews adapter

### PR DESCRIPTION
**Pull request by oleurud**

### CODE REVIEW

File: lib/cartodb/models/mapconfig/adapter/mapconfig-overviews-adapter.js [hunk 1]
```diff
@@ -9,93 +8,94 @@ function MapConfigOverviewsAdapter(overviewsMetadataBackend, filterStatsBackend)
...
+
+    function augmentLayer(layer, done) {
+        if (layer.type !== 'mapnik' && layer.type !== 'cartodb') {
+            return done(null, layer);
+        }
+        self.overviewsMetadataBackend.getOverviewsMetadata(user, layer.options.sql, function (err, metadata) {
+            if (err) {
+                done(err, layer);
+            } else {
+                var query_rewrite_data = { overviews: metadata };
```
Comment (enekid) [inline lib/cartodb/models/mapconfig/adapter/mapconfig-overviews-adapter.js]: I would refactor it to something like:
```javascript
if (err) {
    return done(err, layer)
}

var query_rewrite_data =  { overviews: metadata };
```

Comment (oleurud) [reply lib/cartodb/models/mapconfig/adapter/mapconfig-overviews-adapter.js]: :+1:

**Comment by dgaubert** [comment]
Please, @oleurud. Could you do some DR of what I changed??

File: lib/cartodb/models/mapconfig/adapter/mapconfig-overviews-adapter.js [hunk 2]
```diff
@@ -9,93 +8,128 @@ function MapConfigOverviewsAdapter(overviewsMetadataBackend, filterStatsBackend)
...
-      augmentLayersQueue.defer(augmentLayer, layer);
-  });
-  augmentLayersQueue.awaitAll(layersAugmentQueueFinish);
+        requestMapConfig.layers = layers;
 
+        return callback(null, requestMapConfig);
+    });
 };
+
+function augmentLayer(overviewsMetadataBackend, filterStatsBackend, user, layer, analysesResults, callback) {
```
[the remark this answers is not in the document]
Comment (dgaubert) [reply lib/cartodb/models/mapconfig/adapter/mapconfig-overviews-adapter.js]: 👍 

it was in my mind but at some point I missed it. :)

**Pull request merged**